### DISCUSSION TITLE
Enforcing cannot be changed on a field

### DISCUSSION THREAD

**Discussion by hagen31** [sadieblue/agilityblue.workspace-scripts#21] [Q&A] [answered]
Does AB support a state or read only command? I am trying to write a script to get the value of the matter.reference field when a user opens it then compare the matter.reference field when a user closes the form. I want to ensure a user does not change the value. A read only setting would also work; however, not had any success getting AB to recognize either.

**Comment by bmward80** [comment]
Hi, @hagen31

No, there is no “On read” events. Scripts only support create, update, and delete. Events for reading would be too chatty and cause sluggish behavior in the app. 

You could instead try an on before update event and compare the saved value of the matter reference field and compare it to the incoming matter’s reference field.  If the values are different, either deny the save or replace the reference with the stored one. This would accomplish your goal. 

Brandon

**Reply by hagen31** [reply]
Hello Brandon,

I'll go back to trying that as have been unsuccessful to date.

**Reply by bmward80** [reply, marked as answer]
Do you have an example of your unsuccessful attempts?

You should be able to use the following to get an idea:

```powershell
$incomingMatter = Get-InboundObjectInstance

$storedMatter = Get-Matter -Id $incomingMatter.MatterId

if ($incomingMatter.Reference -ne $storedMatter.Reference) {
    Write-Output "User is attempting to change the reference from '$($storedMatter.Reference)' to '$($incomingMatter.Reference)'"
    
    return $null
}

return $incomingMatter
```

Brandon

**Reply by hagen31** [reply]
Sometimes simplicity is best. I was making this more complicated and was so focused on matter.reference that not having the second line was the key that unlocked. Thank you and this works perfectly.
